src/Lykke.Service.BlockchainWallets.Services/ConstantsService.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Lykke.Service.BlockchainWallets.Contract;
+using Lykke.Service.BlockchainWallets.Core.DTOs;
+using Lykke.Service.BlockchainWallets.Core.Services;
+
+namespace Lykke.Service.BlockchainWallets.Services
+{
+    public class ConstantsService : IConstantsService
+    {
+        private readonly IBlockchainIntegrationService _blockchainIntegrationService;
+        private readonly IDictionary<string, AddressExtensionConstantsDto> _cache;
+
+        public ConstantsService(
+            IBlockchainIntegrationService blockchainIntegrationService)
+        {
+            _blockchainIntegrationService = blockchainIntegrationService;
+            _cache = new Dictionary<string, AddressExtensionConstantsDto>();
+        }
+
+
+        public async Task<AddressExtensionConstantsDto> GetAddressExtensionConstantsAsync(string blockchainType)
+        {
+            var apiClient = _blockchainIntegrationService.TryGetApiClient(blockchainType);
+            if (apiClient == null)
+            {
+                throw new NotSupportedException($"Blockchain type [{blockchainType}] is not supported.");
+            }
+
+            if (!_cache.ContainsKey(blockchainType))
```

Comment:
This code is not thread safe but can be called from multiply threads

Reply:
Fixed